src/zmalloc.c

```diff
@@ -170,6 +198,22 @@ void *zrealloc(void *ptr, size_t size) {
 #endif
 }
 
+void *zrealloc(void *ptr, size_t size) {
+#ifdef USE_MEMKIND
+    void* new_ptr = NULL;
+    if (ptr) {
+        new_ptr = zrealloc_local((void*)((char*)(ptr) - MEMKIND_PREFIX_SIZE), size + MEMKIND_PREFIX_SIZE);
```

Comment:
Is it possible to make use of the same logic in zfree here?  
```
uint64_t *is_ram = (uint64_t*)((char*)(ptr) - MEMKIND_PREFIX_SIZE);
if (*is_ram) {
  zrealloc_local()
} else {
  mrealloc()
}
```

Reply:
Please see newest commit (https://github.com/pmem/redis/pull/86/commits/5256136524c2ca1e336e6e4c62d2cfb11d1106c0)